src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -275,6 +285,24 @@ public MessageBuilder appendTestSummary() {
             return this;
         }
 
+        public MessageBuilder appendCustomMessage() {
+            AbstractProject<?, ?> project = build.getProject();
+            String customMessage = Util.fixEmpty(project.getProperty(SlackNotifier.SlackJobProperty.class)
+                    .getCustomMessage());
+            Map<String, String> buildVariables = new HashMap();
+            buildVariables.putAll(build.getBuildVariables());
+            try {
+                buildVariables.putAll(build.getEnvironment(new LogTaskListener(logger, INFO)));
+            } catch (IOException e) {
+                logger.log(SEVERE, e.getMessage(), e);
+            } catch (InterruptedException e) {
+                logger.log(SEVERE, e.getMessage(), e);
+            }
+            message.append("\n");
+            message.append(replaceMacro(customMessage, new VariableResolver.ByMap<String>(buildVariables)));
```

Comment:
What about using EnvVars `expand()` ? https://github.com/jenkinsci/jenkins/blob/76adbf922369fae214cf86445d422a424f7d309a/core/src/main/java/hudson/EnvVars.java#L382
If I understand it correctly, would also allow usage of Environment Variables?